source/vir/src/modes.rs

```diff
@@ -699,7 +699,16 @@ fn check_expr_handle_mut_arg(
                         if function.x.attrs.atomic {
                             ai.add_atomic(&expr.span);
                         } else {
-                            ai.add_non_atomic(&expr.span);
+                            // A call to `create_open_invariant_credit` is a no-op, so
+                            // it's fine to include in an atomic block. And it's useful
+                            // to be able to do so, so that we can nest an opening of
+                            // an invariant inside an opening of another invariant. So
+                            // we special-case this call to not treat it as non-atomic.
+                            if function.x.name
+                                != crate::fun!("vstd" => "invariant", "create_open_invariant_credit")
```

Comment:
What's the `crate::fun!` macro?

Could this use `verus_items` instead to match this function?

Reply:
I don't seem to be able to use `verus_items` from `vir`. This is consistent with what I thought I learned in the architecture part of the Verus all-hands meeting: that `vir` is at a lower layer than `rust_verify`. Is there a trick to using `rust_verify` functions from `vir`?

Reply:
That's correct, but I'd rather we'd avoid further ad-hoc string-matching in VIR. The common strategy is to still add them to `verus_items`, in the `Vstd` group, and give them a path (as required by the `Vstd` constructor): they should get encoded with that path, which you can then refer to like in this case: https://github.com/verus-lang/verus/blob/f2ed4b35dcd7fd7ad5876d44fa4f1f778deb405e/source/vir/src/def.rs#L782-L790

This way name changes in vstd are less likely to change how things are mapped in VIR.

Let me know if I can help / clarify further.

Reply:
In `vir`, where can I get the `vstd_crate_name` to pass to this? The `ctxt` I have doesn't seem to have a `vstd_crate_name` field.

Reply:
It's in the VIR `GlobalCtx`, you may need to add it to the `ctxt` for prune if it's not already reachable from there. I have little time today so I only gave a cursory look at that, let me know if it doesn't make sense / I can assist more.

Reply:
OK, thanks for the tip. I've addressed this issue in the latest commit.